includes/phonebook-functions.php

```diff
@@ -0,0 +1,425 @@
+				}
+			}
+
+			ksort( $result->staff );
+		}
+
+		if ( $is_org ) {
+			$organizations[] = $result;
+			unset( $results[$key] );
+		}
+
+		if ( $is_dept ) {
+			$departments[] = $result;
+			unset( $results[$key] );
+		}
+	}
+}
+
+/**
+ * Deduplicate staff records with multiple emails
+ * @author Jim Barnes
+ * @since 3.0.0
+ * @param $results Array | The object of results
+ * @return Array | The deduplicated array of results
+ **/
+function phonebook_deduplicate_staff( $results ) {
+	foreach( $results as $key => $result ) {
+		if ( $result->from_table === 'staff' ) {
+			foreach( $results as $_key => $_result ) {
+				// Deduplicate on email
+				if (
+					( $result->email !== null ) &&
+					( $_result->email !== null ) &&
+					( $result !== $_result ) &&
+					( $result->email === $_result->email )
+				) {
+					$_result->secondary[] = $result;
+					unset( $results[$key] );
+				}
+			}
+		}
+	}
+
+	return $results;
+}
+
+/**
+ * Determines if the result is a member of an organization or department
+ * @author Jim Barnes
+ * @since 3.0.0
+ * @param $group Object | The group result
+ * @param $result Object | The staff result
+ * @param $org boolean | True if organization, false if department
+ **/
+function phonebook_result_is_in_group( $group, $result, $org ) {
+	if ( $org ) {
+		return ( $group->name === $result->organization );
+	}
+
+	// This is not an org, check department
+	return ( $group->name === $result->department );
+}
+
+/**
+ * Determines of the result is something we understand
+ * based on what table is comes from.
+ * @author Jim Barnes
+ * @since 3.0.0
+ * @param $result Object | The phonebook result
+ * @return boolean
+ **/
+function phonebook_filter_tables( $result ) {
+	return in_array( $result->from_table, array(
+		"organizations",
+		"departments",
+		"staff"
+	) );
+}
+
+/**
+ * Determines if the word 'fax' is in the result name
+ * @author Jim Barnes
+ * @since 3.0.0
+ * @param $result Object | The phonebook result
+ * @return boolean
+ **/
+function phonebook_filter_fax_in_name( $result ) {
+	return (preg_match( "/^fax\s/i", $result->name ) ||
+		   preg_match( "/\sfax\s/i", $result->name ) ||
+		   preg_match( "/\sfax$/i", $result->name) ) ? false : true;
+}
+
+/**
+ * Formats the result for output to the page.
+ * @author Jim Barnes
+ * @since 3.0.0
+ * @param $result Array | The result from the search service
+ * @return string | The html output
+ **/
+function format_phonebook_result( $result ) {
+	$is_org = ( $result->from_table === 'organizations' );
+	$is_dept = ( $result->from_table === 'departments' );
+	$is_group = ( $is_org || $is_dept );
+
+	$unique_slug = sanitize_title( str_replace( ' ', '-', $result->name ) . $result->id );
+
+	$class = $is_group ? ' group-result' : '';
+
+	ob_start();
+?>
+	<li class="result<?php echo $class; ?> list-unstyled mb-2">
+		<table class="table">
+			<tbody>
+				<tr class="row mx-auto">
+					<td class="col-sm-6">
```

Comment:
I'd recommend bumping table columns up to .col-md-X columns to help prevent long emails/phone numbers from overflowing at the -sm breakpoint.

Adding `text-overflow: ellipsis` to emails and phone numbers would also help prevent overflow.